zjit/src/hir.rs

```diff
@@ -1717,8 +1723,35 @@ fn filter_translatable_calls(flag: u32) -> Result<(), ParseError> {
     Ok(())
 }
 
+#[derive(Debug)]
+struct ProfileOracle {
```

Comment:
It felt difficult to guess why this struct exits by looking at the name "profile oracle" and this definition. Could you add a comment above this struct to clarify the reason why this is necessary besides `IseqPayload::opnd_types`?

My understanding is that `IseqPayload::opnd_types` operates on ISEQ-level indexes whereas `ProfileOracle` translates them into HIR-level indexes, `InsnId`. But it's hard to get that idea from the name `ProfileOracle`. I'm also not familiar with how the word "oracle" is used in the compilers field in general.

Reply:
Thanks for adding the comment :)

Reply:
Oracle largely means "mystical thing that holds the answers" (a reference to Greek mythology, I think), and is a fairly common term in CS theory. I am happy to swap out for a different name